pkg/cloudprovider/vsphere/instances_v2_test.go

```diff
@@ -0,0 +1,591 @@
+				return false, errors.New("provider error")
+			},
+			expected:    false,
+			expectError: true,
+		},
+		{
+			name: "error getting provider ID",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node",
+				},
+				Spec: v1.NodeSpec{
+					ProviderID: "",
+				},
+			},
+			instanceIDFunc: func(ctx context.Context, nodeName types.NodeName) (string, error) {
+				return "", errors.New("instance ID lookup failed")
+			},
+			expected:    false,
+			expectError: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			instances := &mockInstances{
+				instanceIDFunc:                 tc.instanceIDFunc,
+				instanceExistsByProviderIDFunc: tc.instanceExistsByProviderIDFunc,
+			}
+			zones := &mockZones{}
+
+			c := &instancesV2{
+				instances: instances,
+				zones:     zones,
+			}
+
+			ctx := context.Background()
+			result, err := c.InstanceExists(ctx, tc.node)
+
+			if tc.expectError && err == nil {
+				t.Error("expected error but got nil")
+			}
+
+			if !tc.expectError && err != nil {
+				t.Errorf("expected no error but got: %v", err)
+			}
+
+			if result != tc.expected {
+				t.Errorf("expected %t but got %t", tc.expected, result)
+			}
+		})
+	}
+}
+
+func TestInstanceShutdown(t *testing.T) {
+	testCases := []struct {
+		name                             string
+		node                             *v1.Node
+		instanceIDFunc                   func(ctx context.Context, nodeName types.NodeName) (string, error)
+		instanceShutdownByProviderIDFunc func(ctx context.Context, providerID string) (bool, error)
+		expected                         bool
+		expectError                      bool
+	}{
+		{
+			name: "instance is shutdown",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node",
+				},
+				Spec: v1.NodeSpec{
+					ProviderID: "vsphere://instance-123",
+				},
+			},
+			instanceShutdownByProviderIDFunc: func(ctx context.Context, providerID string) (bool, error) {
+				return true, nil
+			},
+			expected:    true,
+			expectError: false,
+		},
+		{
+			name: "instance is not shutdown",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node",
+				},
+				Spec: v1.NodeSpec{
+					ProviderID: "vsphere://instance-123",
+				},
+			},
+			instanceShutdownByProviderIDFunc: func(ctx context.Context, providerID string) (bool, error) {
+				return false, nil
+			},
+			expected:    false,
+			expectError: false,
+		},
+		{
+			name: "error checking instance shutdown",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node",
+				},
+				Spec: v1.NodeSpec{
+					ProviderID: "vsphere://instance-123",
+				},
+			},
+			instanceShutdownByProviderIDFunc: func(ctx context.Context, providerID string) (bool, error) {
+				return false, errors.New("provider error")
+			},
+			expected:    false,
+			expectError: true,
+		},
+		{
+			name: "error getting provider ID",
```

Comment:
name should be "error getting instance ID"?

Reply:
ah good catch.  I'll fix that.